src/robust.jl

```diff
@@ -44,10 +44,10 @@ function trim!(x::AbstractVector; prop::Real=0.0, count::Integer=0)
         0 <= count < n/2 || throw(ArgumentError("count must satisfy 0 ≤ count < length(x)/2."))
     end
 
-    partialsort!(x, (n-count+1):n)
-    partialsort!(x, 1:count)
-    deleteat!(x, (n-count+1):n)
-    deleteat!(x, 1:count)
+    ix = vcat(partialsortperm(x, 1:count), partialsortperm(x, (n-count+1):n))
+    sort!(ix)
+    ix = unique(ix) # can be replaced by unique! starting julia 1.1
```

Comment:
Since you know indices are sorted, a simple loop comparing values to previous ones and skipping them will probably be faster.

Reply:
Have you tried this?